libkbfs/disk_block_cache.go

```diff
@@ -411,29 +400,66 @@ func (cache *DiskBlockCacheStandard) Put(ctx context.Context, tlfID tlf.ID,
 		return err
 	}
 	if !hasKey {
-		for cache.currBytes+uint64(encodedLen) > cache.maxBytes {
-			cache.evictLocked(ctx, defaultNumBlocksToEvict)
+		for {
+			select {
```

Comment:
Sorry, I forgot to mention this last review, but I'm a bit nervous that we can get into a state (e.g. due to a bug) where the limiter tells us to evict, but we can't evict any more, and so we'd burn the CPU until the context times out. Perhaps check whether `evictLocked` didn't actually do anything, and treat that case as an error?